static/app/utils/performanceForSentry.tsx

```diff
@@ -0,0 +1,26 @@
+import {timestampWithMs} from '@sentry/utils';
+
+import getCurrentSentryReactTransaction from './getCurrentSentryReactTransaction';
+
+const MIN_UPDATE_SPAN_TIME = 16; // Frame boundary @ 60fps
+
+/**
+ * Callback for React Profiler https://reactjs.org/docs/profiler.html
+ */
+export function onRenderCallback(
+  id: string,
+  phase: 'mount' | 'update',
+  actualDuration: number
+) {
+  const transaction = getCurrentSentryReactTransaction();
+  if (transaction && actualDuration > MIN_UPDATE_SPAN_TIME) {
+    const now = timestampWithMs();
+    const span = transaction.startChild({
+      description: `<${id}>`,
+      op: `ui.react.${phase}`,
+      startTimestamp: now - actualDuration / 1000,
+      endTimestamp: now,
+    });
+    span.finish();
```

Comment:
You don't have to call `span.finish` if you give an `endTimestamp`.

Reply:
Oh really? TIL